### PR TITLE
Json value filter improvements

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: model/src/main/kotlin/utils/JsonIncludeValueFilters.kt [hunk 1]
```diff
@@ -19,44 +19,45 @@
 
 package org.ossreviewtoolkit.model.utils
 
+import kotlin.reflect.KClass
+import kotlin.reflect.safeCast
```
Comment (sschuberth) [inline model/src/main/kotlin/utils/JsonIncludeValueFilters.kt:23]: I'd prefer the model to stay free of reflection, which helps with GraalVM native-images and compiling to non-JVM platforms. Can we simply drop this commit?

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/utils/JsonIncludeValueFilters.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/utils/JsonIncludeValueFilters.kt:23]: @mnonnenmacher @MarcelBochtler  does this align with your view as well?

Comment (mnonnenmacher) [reply model/src/main/kotlin/utils/JsonIncludeValueFilters.kt]: Looks like you removed the commit already so I can't give a specific answer, but IMO we should generally try to keep our Jackson usage simple to not hinder the planned migration to Kotlin serialization.

Comment (sschuberth) [reply model/src/main/kotlin/utils/JsonIncludeValueFilters.kt]: > Looks like you removed the commit already so I can't give a specific answer

You can still [see it in the diff of the force-push](https://github.com/oss-review-toolkit/ort/compare/38e5c14ca64bb944f509cfd3e70c893f96e8b524..1c442a49c545c675b96045326a02ee405f6e06b0).

**Review by sschuberth — APPROVED** [review]

**Pull request merged**